app/routes/circulars._archive._index/route.tsx

```diff
@@ -170,6 +171,18 @@ export default function () {
   const formId = useId()
   const submit = useSubmit()
 
+  const dateSelectorRef = useRef<HTMLDivElement>(null)
+  const [showDateSelector, setShowDateSelector] = useState(false)
+  const SortSelectorRef = useRef<HTMLDivElement>(null)
+  const [showSortSelector, setShowSortSelector] = useState(false)
+
+  useOnClickOutside(dateSelectorRef, () => {
+    setShowDateSelector(false)
+  })
+  useOnClickOutside(SortSelectorRef, () => {
+    setShowSortSelector(false)
+  })
```

Comment:
These refs and states are only used inside `SortSelector` and `DateSelector`. Please move them inside those two respective components. They need not be exposed as props.